feat(icon): convert icon components to ts

EdwardIrby opened the pull request:
Icon ts conversion

jaketrent commented on packages/icon/cli/generate-components.js:
```diff
@@ -7,23 +7,35 @@ const svgo = new SVGO({ plugins: [{ removeXMLNS: true }] })
-${componentName}.propTypes = {
-  'aria-label': PropTypes.string,
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
```
Something like `interface ${capitalize(componentName)}Props` would be good here.

jaketrent commented on packages/icon/src/css/index.ts:
```diff
@@ -6,7 +6,7 @@ import {
 } from '@pluralsight/ps-design-system-core'
 
-import * as vars from '../vars/index.js'
+import * as vars from '../vars/'
```
nit

```suggestion
import * as vars from '../vars'
```

jaketrent commented on packages/icon/src/react/index.tsx:
```diff
@@ -0,0 +1,55 @@
+import { css } from 'glamor'
```
Really wish we could get a workflow where I could see diffs compared to the last file instead of this situation where git thinks this is a brand new file. 

Currently:
I lose (most recent) git blame. It's way harder to review.

jaketrent reviewed (commented):
yarn build and tsc are happy.

Please convert the test and stories before merging.  I know you've been reluctant to do this so far. If you want to push for the whole project to not do this, we should have a team discussion before we go on.

My preference is that we do a full conversion to TS.

There is already a good precedent for this in other conversions:

```
dev/design-system/packages on  feat/icon-ts-conversion [$] ✦ ❯ fd spec.ts **/src
avatar/src/js/__specs__/index.spec.ts
avatar/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap
avatar/src/react/__specs__/index.spec.tsx
avatar/src/react/__specs__/storyshots.spec.ts
avatar/src/vars/__specs__/keymirror.spec.ts
emptystate/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap
emptystate/src/react/__specs__/index.spec.tsx
emptystate/src/react/__specs__/storyshots.spec.ts
screenreaderonly/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap
screenreaderonly/src/react/__specs__/index.spec.tsx
screenreaderonly/src/react/__specs__/storyshots.spec.ts
util/src/__specs__/debounce.spec.ts
util/src/menu-keys/__specs__/index.spec.tsx
dev/design-system/packages on  feat/icon-ts-conversion [$] ✦ ❯ fd story.ts **/src
avatar/src/react/__stories__/index.story.tsx
circularprogress/src/react/__stories__/index.story.tsx
emptystate/src/react/__stories__/index.story.tsx
screenreaderonly/src/react/__stories__/index.story.tsx
tab/src/react/__stories__/index.story.tsx
util/src/menu-keys/__stories__/index.story.tsx
```

I want to talk about this instead of going forward with a mixed conversion..

jaketrent reviewed (changes requested):
Please upgrade the string to more specific types as indicated inline.  Then, a few more things:

Still a chunk of js files that need converted:

```
❯ fd js src
src/react/__specs__/__snapshots__/storyshots.spec.js.snap
src/react/__specs__/index.spec.js
src/react/__specs__/storyshots.spec.js
src/react/__stories__/index.story.js
```

One failing test:

```
 FAIL   @pluralsight/ps-design-system-icon  src/react/__specs__/index.spec.js
  ● Test suite failed to run

    Cannot find module '../index.js' from 'AccountIcon.dist.js'
```

I would also suggest turning `strict: true` on in your tsconfig.json.  We haven't been doing this, and I think we should on all conversions.

Let's chat about any of this as needed.

EdwardIrby commented on packages/icon/src/react/index.tsx:
```diff
@@ -0,0 +1,56 @@
+}
+
+export interface IconProps extends HTMLAttributes<HTMLDivElement> {
+  size?: string
```
You need to clean up your directory those are ignored files

EdwardIrby commented:
> Please upgrade the string to more specific types as indicated inline. Then, a few more things:
> 
> Still a chunk of js files that need converted:
> 
> ```
> ❯ fd js src
> src/react/__specs__/__snapshots__/storyshots.spec.js.snap
> src/react/__specs__/index.spec.js
> src/react/__specs__/storyshots.spec.js
> src/react/__stories__/index.story.js
> ```
> 
> One failing test:
> 
> ```
>  FAIL   @pluralsight/ps-design-system-icon  src/react/__specs__/index.spec.js
>   ● Test suite failed to run
> 
>     Cannot find module '../index.js' from 'AccountIcon.dist.js'
> ```
> 
> I would also suggest turning `strict: true` on in your tsconfig.json. We haven't been doing this, and I think we should on all conversions.
> 
> Let's chat about any of this as needed.

Those are ignored files you need toc clean up your directory though that does give me an idea of what the problem is.

jaketrent reviewed (approved):
Local build looks good.  Cleaned and rebuilt and didn't have js files any more. Added one commit to get the proper enum typings on the sizes and colors.

EdwardIrby commented:
> Local build looks good. Cleaned and rebuilt and didn't have js files any more. Added one commit to get the proper enum typings on the sizes and colors.

This breaks both my local build and test, even after deleting branch and pulling it back down.
